WordPress/gutenberg
# Image/Gallery Blocks: add has-caption class

## What?

Adds a `has-caption` class for the Image and Gallery blocks.

## Why?

Allows specific styling for the figure element when it contains an image caption.

## How?

Adds a css class when a image and/or gallery includes a caption.

## Testing Instructions

- Insert an image block, verify that the `has-caption` class is added to the `figure` element in the editor, when a caption is added
- Insert a gallery block, verify that the `has-caption` class is added to the `figure` element in the editor, when a caption is added
- Save both blocks and verify that the block output includes the class, as well
- Open a post with image and gallery blocks saved before this change, and verify there are no block validation errors in the JS console
